### PR TITLE
Initial commit of fourth JAX-101 notebook

### PR DESCRIPTION
**Pull request by jakevdp**

### CODE REVIEW

**Review by 8bitmp3 — COMMENTED** [review]
Thanks again, this is awesome @jakevdp 👍

I've included a few small suggestions, in case you needed an extra pair of eyes on this long Wednesday:

- 2-3 nit/typo fixes
- Additions of en dashes/hyphens in "higher-order"
- Providing a link to "TD learning" for those folks who aren't familiar with RL: `...TD(0) ([temporal difference](https://en.wikipedia.org/wiki/Temporal_difference_learning)) reinforcement learning...`. (It's similar to how you provide a link here: `[Hessian matrix](https://en.wikipedia.org/wiki/Hessian_matrix)`.)
- Spelling out "MAML" as "Model-Agnostic Meta-Learning (MAML)" (I'm quite familiar with this one—Dr Finn's awesome thesis + paper which inspired Reptile and ANIL).

Cheers ☕

File: docs/jax-101/04-advanced-grads.md [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+
+# Advanced gradients in JAX
+
+[![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/google/jax/blob/master/docs/jax-101/04-advanced-grads.ipynb)
+
+*Authors: TODO*
+
+Computing gradients is a critical part of modern machine learning methods. This section considers a few advanced topics in the areas of automatic differentiation as it relates to modern machine learning.
+
+While understanding how automatic differentiation works under the hood isn't crucial for using JAX in most contexts, we encourage the reader to check out this quite accessible [video](https://www.youtube.com/watch?v=wG_nF1awSSY&t=697s)  to get a deeper sense of what's going on.
```
Comment (skye) [inline docs/jax-101/04-advanced-grads.md:24]: Don't include t=697 in video url

**Review by skye — APPROVED** [review]
FYI @mattjj. You had some additional review comments for this notebook, but I think we should get this out as-is, and you or someone else with sufficient motivation can make the more involved changes. Does this sound good?

File: docs/jax-101/04-advanced-grads.md [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+    format_version: 0.13
+    jupytext_version: 1.10.0
+kernelspec:
+  display_name: Python 3
+  name: python3
+---
+
++++ {"id": "kORMl5KmfByI"}
+
+# Advanced gradients in JAX
```
[the remark this answers is not in the document]
Comment (jakevdp) [reply docs/jax-101/04-advanced-grads.md]: Done, thanks

File: docs/jax-101/04-advanced-grads.md (hunk 1 above)
Comment (jakevdp) [reply docs/jax-101/04-advanced-grads.md]: Done

**Comment by jakevdp** [comment]
@skye @mattjj I wasn't exactly sure how to approach the other comments from the review (expanding the MAML section and rewriting TD(0) without stop_gradient) so I'm leaving that out for now. Should we open an issue to track that?

**Comment by mattjj** [comment]
Yeah, opening an issue would be awesome. Could be one issue for both. Could you open one and assign me, if it's convenient?

**Comment by jakevdp** [comment]
Done: #5935

**Pull request merged**